libmamba/src/core/curl.hpp

```diff
@@ -64,6 +64,38 @@ namespace mamba
         bool m_serious;
     };
 
+    class CURLId
+    {
+    public:
+
+        explicit CURLId(CURL* handle = nullptr);
+
+        bool operator==(const CURLId& rhs) const;
+        bool operator!=(const CURLId& rhs) const;
+        bool operator<(const CURLId& rhs) const;
+        bool operator<=(const CURLId& rhs) const;
+        bool operator>(const CURLId& rhs) const;
+        bool operator>=(const CURLId& rhs) const;
+
+        std::size_t hash() const noexcept;
+
+    private:
+
+        CURL* p_handle;
+    };
+}
+
+template <>
+struct std::hash<mamba::CURLId>
```

Comment:
Not totally sure but I htink `::std::hash` have to be used to make sure it's the standard and not a sub-namesapce.

Reply:
Actually this is not possible, the compiler complains with: `error: global qualification of class name is invalid before '{' token`